tracing-opentelemetry/src/subscriber.rs

```diff
@@ -421,6 +421,22 @@ where
             builder.trace_id = Some(self.tracer.new_trace_id());
         }
 
+        let builder_attrs = builder.attributes.get_or_insert(Vec::new());
+
+        let meta = attrs.metadata();
+
+        if let Some(filename) = meta.file() {
+            builder_attrs.push(KeyValue::new("code.filepath", filename));
+        }
+
+        if let Some(module) = meta.module_path() {
+            builder_attrs.push(KeyValue::new("code.namespace", module));
+        }
+
+        if let Some(line) = meta.line() {
+            builder_attrs.push(KeyValue::new("code.lineno", line as i64));
+        }
```

Comment:
i wonder if we want to only produce this attribute if the `module_path` or `file` values are present? a line number without a file or module name doesn't make a lot of sense.

on the other hand, the `tracing` macros will never generate metadata that has a line number without a filename or module path, and i'd be very surprised if any other instrumentation did that. so, it's not a big deal either way.

Reply:
My thinking was that we should try to be defensive and include the maximum possible amount of information on each span.